test/functional/collection.test.js

```diff
@@ -1307,4 +1312,34 @@ describe('Collection', function() {
       });
     }
   });
+
+  it('should createIndex with serverSelection readPreference primary', {
+    metadata: {
+      requires: { topology: 'replicaset' }
+    },
+    test: function(done) {
+      const configuration = this.configuration;
+      const client = configuration.newClient(configuration.writeConcernMax(), {
+        useUnifiedTopology: true,
+        readPreference: 'primaryPreferred'
+      });
+      client.connect((err, client) => {
+        expect(err).to.not.exist;
+        const db = client.db(configuration.db);
+        const collection = db.collection('db-two');
+        TopologyStub = sinon.spy(Topology.prototype, 'selectServer');
+        collection.createIndex({ quote: 'text' }, (err, res) => {
+          expect(err).to.not.exist;
+          expect(res).to.equal('quote_text');
+          expect(TopologyStub.called).to.equal(true);
+          expect(TopologyStub.args[0][0].readPreference.mode).to.equal(ReadPreference.PRIMARY);
```

Comment:
```suggestion
          expect(TopologyStub).nested.property('args[0][0].readPreference.mode').to.equal(ReadPreference.PRIMARY);
```
prefer using chai's nested `property` assertion helper to better inform error messages when things go wrong:
```